tests/auth.test.ts

```diff
@@ -0,0 +1,65 @@
+import { test, expect } from 'vitest';
+import { gql } from 'graphql-request';
+import { cleanDatabase } from './helpers/database';
+import { ctx, executeGraphQlQuery } from './helpers/server';
+
+const signUpMutation = gql`
+  mutation {
+    signUp(user: { username: "asdf", password: "asdf" }) {
+      user {
+        username
+      }
+      token
+    }
+  }`;
```

Comment:
nexus itself provides examples with hardcodng them
https://nexusjs.org/docs/getting-started/tutorial/chapter-testing-your-ap

i've tried to find something that generates it but did not discover any good looking packages that return a plain string with query there

Reply:
Oh, that's terrible. I wouldn't want to hardcode those things in plaintext, since it looses the point of having typescript and being able to validate that `signUp` even exist in the first place. Some kind of query builder will be a better choice here, I believe. Not sure about https://www.npmjs.com/package/gql-query-builder, maybe there is something better/provided by the nexus/nicely integrated with it?

Reply:
This looks promising https://github.com/remorses/genql

Reply:
used this package. There's still no type check safety, but tests fail if the built query does not correspond to the defined structure. (e.g. if one fails to provide the correct type name for data sent with sign up mutation)

Did not find a better option of a builder.

Reply:
i did not figure out how to handle the errors there without tests behaving incorrectly. Had to come up with a hack `catch(e) => e` in cases where an error is expected.

the behaviour is inconsistent and strange. If i am doing `expect.rejects` then e.g. test with incorrect password sign in fails because it does not reject. I wiggled around and also discovered that even if all the tests pass with `reject`, then there's a global error from vitest claiming that there's an unhandled error. Although i don't think there is one. And on top of all that the error that is thrown does not seem to correspond to a test case that the error points to.

Reply:
> i did not figure out how to handle the errors

Do you describe the current state of the PR after you've integrated [genql](https://github.com/remorses/genql)? Do you suggest to revert back this integration? I see the potential problem in the fact that you use generated files [in the same code where you've generated them](https://github.com/makerdao-ses/switchboard-boilerplate/blob/6aa6420b4df0a36f1b0bd41cfd931f7056970b53/tests/helpers/server.ts#L9). But I'm also not a big fan of the amount of boilerplate code and strange packages (`undici`) that were added to integrate this, so I'm not against of reverting it back

Reply:
> Do you describe the current state of the PR after you've integrated [genql](https://github.com/remorses/genql)?

yes

> Do you suggest to revert back this integration?

yes due to:
  - documentation is not so big and clear
  - the client error handling is not straightforward when combined with vitest
  - we would have to keep 2 generated folders around.

Reply:
Alright. For the next issue, this also looks relevant and have more stars: [graphql-code-generator](https://github.com/dotansimha/graphql-code-generator)